app/forms/provider_interface/reconfirm_deferred_offer_wizard.rb

```diff
@@ -0,0 +1,137 @@
+module ProviderInterface
+  class ReconfirmDeferredOfferWizard
+    include ActiveModel::Model
+
+    STEPS = %w[start conditions check].freeze
+
+    attr_accessor :current_step, :application_choice_id
+    attr_accessor :conditions_status, :course_option_id
+    attr_writer :state_store
+
+    validates :application_choice_id, :current_step, presence: true
+
+    validate :validate_step,
+             :validate_application_choice,
+             :is_course_option_still_available
+
+    validates :conditions_status, presence: true, if: proc { |f|
+      %w[conditions check].include? f.current_step
+    }
```

Comment:
It's slightly more idiomatic to use validation contexts for this: see https://github.com/DFE-Digital/apply-for-teacher-training/blob/15cc371bd7926a46bcc8ff7fd049bb0846591d81/app/forms/provider_interface/provider_user_invitation_wizard.rb#L10 and https://github.com/DFE-Digital/apply-for-teacher-training/blob/15cc371bd7926a46bcc8ff7fd049bb0846591d81/app/forms/provider_interface/provider_user_invitation_wizard.rb#L63-L65

`on:` declares a validation context for each validation in the list, so you can say (as `valid_for_current_step?` does, `valid(:conditions_status)`